pkg/resource/deploy/deployment.go

```diff
@@ -314,6 +314,10 @@ type Deployment struct {
 	newPlans *resourcePlans
 	// the set of resources read as part of the deployment
 	reads *gsync.Map[resource.URN, *resource.State]
+	// a channel to post steps back to the step generator.
+	resourceStatusEvents <-chan SourceEvent
+	// the resource status server.
+	resourceStatus *resourceStatusServer
```

Comment:
This is here out of convenience. It probably makes more sense on the deployment executor than the deployment, but all the existing steps already have a reference to the deployment, so it's convenient to access from here rather than updating all the steps to take in `resourceStatus`.